datafusion/sql/tests/cases/plan_to_sql.rs

```diff
@@ -1746,3 +1749,153 @@ fn test_unparse_subquery_alias_with_table_pushdown() -> Result<()> {
+        .join_on(
+            subquery,
+            datafusion_expr::JoinType::LeftMark,
+            vec![col("t1.id").eq(col("__correlated_sq_1.id"))],
+        )?
+        .project(vec![col("t1.d")])?
+        .filter(col("mark").or(col("t1.d").lt(lit(0))))?
+        .build()?;
+
+    let unparser = Unparser::new(&UnparserPostgreSqlDialect {});
+    let sql = unparser.plan_to_sql(&plan)?;
+    assert_eq!("SELECT \"t1\".\"d\" FROM \"t1\" WHERE (EXISTS (SELECT 1 FROM \"t2\" AS \"__correlated_sq_1\" WHERE (\"t1\".\"id\" = \"__correlated_sq_1\".\"id\")) OR (\"t1\".\"d\" < 0))", sql.to_string());
+    Ok(())
+}
+
+#[test]
+fn test_unparse_right_semi_join() -> Result<()> {
+    let schema = Schema::new(vec![
+        Field::new("c", DataType::Int32, false),
+        Field::new("d", DataType::Int32, false),
+    ]);
+    let left = table_scan(Some("t1"), &schema, Some(vec![0, 1]))?.build()?;
+    let right_table_scan = table_scan(Some("t2"), &schema, Some(vec![0, 1]))?.build()?;
+    let right = LogicalPlanBuilder::from(right_table_scan)
+        .project(vec![col("c"), col("d")])?
+        .build()?;
+    let plan = LogicalPlanBuilder::from(left)
+        .join(
+            right,
+            datafusion_expr::JoinType::RightSemi,
+            (
+                vec![Column::from_qualified_name("t1.c")],
+                vec![Column::from_qualified_name("t2.c")],
+            ),
+            None,
+        )?
+        .filter(col("t1.c").lt_eq(lit(1i64)))?
+        .build()?;
+    let unparser = Unparser::new(&UnparserPostgreSqlDialect {});
+    let sql = unparser.plan_to_sql(&plan)?;
+    assert_eq!("SELECT \"t2\".\"c\", \"t2\".\"d\" FROM \"t1\" WHERE (\"t1\".\"c\" <= 1) AND EXISTS (SELECT 1 FROM \"t1\" WHERE (\"t1\".\"c\" = \"t2\".\"c\"))", sql.to_string());
+    Ok(())
+}
+
+#[test]
+fn test_unparse_right_anti_join() -> Result<()> {
+    let schema = Schema::new(vec![
+        Field::new("c", DataType::Int32, false),
+        Field::new("d", DataType::Int32, false),
+    ]);
+    let left = table_scan(Some("t1"), &schema, Some(vec![0, 1]))?.build()?;
+    let right_table_scan = table_scan(Some("t2"), &schema, Some(vec![0, 1]))?.build()?;
+    let right = LogicalPlanBuilder::from(right_table_scan)
+        .project(vec![col("c"), col("d")])?
+        .build()?;
+    let plan = LogicalPlanBuilder::from(left)
+        .join(
+            right,
+            datafusion_expr::JoinType::RightAnti,
+            (
+                vec![Column::from_qualified_name("t1.c")],
+                vec![Column::from_qualified_name("t2.c")],
+            ),
+            None,
+        )?
+        .filter(col("t1.c").lt_eq(lit(1i64)))?
+        .build()?;
+    let unparser = Unparser::new(&UnparserPostgreSqlDialect {});
+    let sql = unparser.plan_to_sql(&plan)?;
+    assert_eq!("SELECT \"t2\".\"c\", \"t2\".\"d\" FROM \"t1\" WHERE (\"t1\".\"c\" <= 1) AND NOT EXISTS (SELECT 1 FROM \"t1\" WHERE (\"t1\".\"c\" = \"t2\".\"c\"))", sql.to_string());
```

Comment:
This result isn't valid for DataFusion and Postgres. 
```
psql=# create table t1 (c int, d int);
CREATE TABLE
psql=# create table t2 (c int, d int);
CREATE TABLE
psql=# SELECT "t2"."c", "t2"."d" FROM "t1" WHERE ("t1"."c" <= 1) AND NOT EXISTS (SELECT 1 FROM "t1" WHERE ("t1"."c" = "t2"."c"));
ERROR:  missing FROM-clause entry for table "t2"
LINE 1: SELECT "t2"."c", "t2"."d" FROM "t1" WHERE ("t1"."c" <= 1) AN...
```
I guess the clause `FROM t1` is wrong.